### PR TITLE
move obo urls mapping to dataload

### PR DESCRIPTION
**Pull request by jamesamcl**

### CODE REVIEW

**Comment by jamesamcl** [comment]
#167

File: dataload/owl2json/src/main/java/uk/ac/ebi/owl2json/xrefs/Bioregistry.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        try {
+            theRegistry = urlToJson(jsonUrl).getAsJsonObject();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+
+        for(var entry : theRegistry.entrySet()) {
+            JsonObject db = entry.getValue().getAsJsonObject();
+            prefixToDatabase.put(db.get("preferred_prefix").getAsString(), db);
+        }
```
Comment (cthoyt) [inline dataload/owl2json/src/main/java/uk/ac/ebi/owl2json/xrefs/Bioregistry.java:58]: Add in indexing of synonyms (needs some checking if there's no `synonyms` entry, but i'm not really much of a java programmer these days)

```suggestion
           for(JsonElement synonym : db.get("synonyms").getAsJsonArray())
               prefixToDatabase.put(synonym.getAsString(), db)
        }
```

File: dataload/owl2json/src/main/java/uk/ac/ebi/owl2json/xrefs/Bioregistry.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        if(patternObj == null)
+            return null;
+
+        Pattern pattern = getPattern(patternObj.getAsString());
+
+        if(!pattern.matcher(id).matches()) {
+            return null;
+        }
+
+        return db.get("uri_format").getAsString().replace("$1", id);
```
Comment (cthoyt) [inline dataload/owl2json/src/main/java/uk/ac/ebi/owl2json/xrefs/Bioregistry.java]: should check that uri_format is not null - not all records in bioregistry have a uri_format

Comment (jamesamcl) [reply dataload/owl2json/src/main/java/uk/ac/ebi/owl2json/xrefs/Bioregistry.java]: Thanks, done!

**Pull request merged**